src/pyhf/probability.py

```diff
@@ -1,40 +1,54 @@
 from . import get_backend
 
 
-class Poisson(object):
+class _SimpleDistributionMixin(object):
+    def log_prob(self, value):
+        return self._pdf.log_prob(value)
+
+    def expected_data(self):
+        return self._pdf.expected_data()
+
+    def sample(self, sample_shape=()):
+        return self._pdf.sample(sample_shape)
+
+
+class Poisson(_SimpleDistributionMixin):
     def __init__(self, rate):
+        tensorlib, _ = get_backend()
         self.rate = rate
+        self._pdf = tensorlib.poisson_pdfcls(rate)
```

Comment:
I realize this is going to be used to assign a private attribute, but I would suggest a more clear name than `poisson_pdfcls`. Maybe I'm missing something very obvious, but why the `cls`?

Reply:
we have:
* `poisson_logpdf`
* `poisson`
* `poisson_pdfcls`

so I was concerned `poisson_pdf` would be confused to be the non-log version of `poisson_logpdf` and return a number. Instead `poisson_pdfcls` returns a instance of a Poisson class

Reply:
This is what I assumed, and I think it is reasonable to indicate that this is the pdf class. `cls` is okay, though I might be more inclined to just name the method `poisson_dist`, though I will concede that this no longer explicitly has `pdf` in the name.

Reply:
I like `dist`